Lib/importlib/_bootstrap.py

```diff
@@ -85,6 +85,10 @@ def __enter__(self):
     def __exit__(self, *args, **kwargs):
         """Remove self.lock from this thread's _blocking_on list."""
         self.blocked_on.remove(self.lock)
+        if len(self.blocked_on) == 0:
+            # gh-101766: glboal cache should be cleaned-up
+            # if there is no more _blocking_on for this thread.
+            del _blocking_on[self.thread_id]
```

Comment:
Would it be worth completely undoing what `__enter__()` did?

```suggestion
            del _blocking_on[self.thread_id]
            del self.blocked_on
```